docker/build.bzl

```diff
@@ -68,23 +102,26 @@ def _build_layer(ctx):
         dirname(ctx.outputs.out.short_path),
         _canonicalize_path(ctx.attr.data_path))
     args += ["--file=%s=%s" % (f.path, strip_prefix(f.short_path, data_path))
-             for f in ctx.files.files]
+             for f in files]
   else:
     # Otherwise, files are added without a directory prefix at all.
     args += ["--file=%s=%s" % (f.path, f.basename)
-             for f in ctx.files.files]
+             for f in files]
 
   args += ["--tar=" + f.path for f in ctx.files.tars]
   args += ["--deb=" + f.path for f in ctx.files.debs if f.path.endswith(".deb")]
-  args += ["--link=%s:%s" % (k, ctx.attr.symlinks[k])
-           for k in ctx.attr.symlinks]
+  for k in symlinks:
+    if ':' in k:
+      fail("The source of a symlink cannot contain ':', got: %s" % k)
```

Comment:
Because of how we pass it to the underlying binary.
  "foo:bar": "baz"

Would be passed indistinguishably from:
  "foo": "bar:baz"